static_test_driver/autosequence.ino

```diff
@@ -139,20 +141,24 @@ void run_control() {
       }
       // Check that the engine is producing thrust once mainstage valves are fully open
 #if CONFIGURATION != DEMO
-      else if (run_time >= STARTUP_TIME && force < MIN_THRUST) {
+      else if (run_time >= THRUST_CHECK_TIME && force < MIN_THRUST) {
 #else
-      else if (force < MIN_THRUST) {
+      else if (run_time >= THRUST_CHECK_TIME && force < MIN_THRUST) {
 #endif
         Serial.println(F("Thrust below critical level"));
         abort_autosequence();
       }
 
-      if (run_time >= RUN_TIME) {
+      //this checks 0.5 sec before runtime is up that the oxygen valves are open and closes them, and doesn't send more commands to close them once they are closed.
+      if (run_time >= RUN_TIME-OX_LEADTIME && valve_status[OXY_PRE] == 1 && valve_status[OXY_MAIN] == 1) {
```

Comment:
This should really be a new state (maybe call it SHUTDOWN?) between MAINSTAGE and COOL_DOWN.

Reply:
I agree but it was easier to write this check in for testing purposes. It should be fine for the test this weekend

Reply:
Hmm, I suppose... I guess if there is an abort both will close at the same time, but that might not be a problem.